includes/specials/SpecialNewpages.php

```diff
@@ -550,25 +457,123 @@ function getQueryInfo() {
 		return $info;
 	}
 
-	function getIndexField() {
-		return 'rc_timestamp';
-	}
-
-	function formatRow( $row ) {
-		return $this->mForm->formatRow( $row );
-	}
-
-	function getStartBody() {
+	protected function preprocessResults( $result ) {
 		# Do a batch existence check on pages
 		$linkBatch = new LinkBatch();
 		foreach ( $this->mResult as $row ) {
-			$userName = User::getUsername( $row->rc_user, $row->rc_user_text ); // SUS-812
+			$userId = $row->rc_user;
+			$userName = User::getUsername( $userId, inet_ntop( $row->rc_ip_bin ) ); // SUS-812
 
 			$linkBatch->add( NS_USER, $userName );
 			$linkBatch->add( NS_USER_TALK, $userName );
 			$linkBatch->add( $row->rc_namespace, $row->rc_title );
+
+			// SUS-3079: Cache the user name
+			if ( $userId && !isset( $this->userNames[$userId] ) ) {
+				$this->userNames[$userId] = $userName;
+			}
 		}
+
 		$linkBatch->execute();
+	}
+
+	function getIndexField() {
+		return 'rc_timestamp';
+	}
+
+
+	/**
+	 * Format a row, providing the timestamp, links to the page/history, size, user links, and a comment
+	 *
+	 * @param object $result Result row
+	 * @return String
+	 */
+	function formatRow( $result ) {
+		# Revision deletion works on revisions, so we should cast one
+		$row = array(
+			'comment' => $result->rc_comment,
+			'deleted' => $result->rc_deleted,
+			// SUS-3079: Use user name lookup here
+			'user_text' => $this->userNames[$result->rc_user] ?? inet_ntop( $result->rc_ip_bin ),
+			'user' => $result->rc_user,
+		);
+		$rev = new Revision( $row );
+
+		$classes = array();
+
+		$lang = $this->getLanguage();
+		$dm = $lang->getDirMark();
+
+		$title = Title::newFromRow( $result );
+		$spanTime = Html::element( 'span', array( 'class' => 'mw-newpages-time' ),
+			$lang->timeanddate( $result->rc_timestamp, true )
+		);
+		$time = Linker::linkKnown(
+			$title,
+			$spanTime,
+			array(),
+			array( 'oldid' => $result->rc_this_oldid ),
+			array()
+		);
+
+		$query = array( 'redirect' => 'no' );
+
+		if( $this->patrollable( $result ) ) {
+			$query['rcid'] = $result->rc_id;
+		}
+
+		$plink = Linker::linkKnown(
+			$title,
+			null,
+			array( 'class' => 'mw-newpages-pagename' ),
+			$query,
+			array( 'known' ) // Set explicitly to avoid the default of 'known','noclasses'. This breaks the colouration for stubs
+		);
+		$histLink = Linker::linkKnown(
+			$title,
+			wfMsgHtml( 'hist' ),
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Replace this legacy wfMsg* function call with one of the new i18n functions. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarqube.wikia-inc.com/coding_rules#rule_key=wikia-php-rules%3ALegacyWfMsgUsage)

[project-id]: wikia:app